jobs/video_jobs/maya_submit.sh.erb

```diff
@@ -12,62 +12,118 @@ BIND_DIRS="$JOB_TMPDIR:/tmp"
 # set a new tmpdir bc that's what the container needs to look at
 export TMPDIR="/tmp"
 
-function create_thumbnails(){
-  RESIZE="-resize 80x80"
-  #IMG_EXTRA_ARGS="-auto-gamma"
-  IMG_ARGS="$RESIZE $IMG_EXTRA_ARGS"
-
-  IMG_DIR="<%= project_dir %>/images"
-  THUMB_DIR="<%= project_dir %>/thumbnails"
-
-  mkdir $THUMB_DIR 2>/dev/null
-
-  for IMG in $(ls <%= project_dir %>/images 2>/dev/null)
-  do
-    BASE_IMG_NAME=${IMG##*/}
-    IMG_NAME=${BASE_IMG_NAME%.*}
-
-    convert $IMG_DIR/$IMG $IMG_ARGS $THUMB_DIR/$IMG_NAME.png
-  done
-}
-
-function start_frame(){
-  if [ -z "$PBS_ARRAYID" ]; then
-    PBS_ARRAYID=1
-  fi
-  
-  if [ $PBS_ARRAYID -gt 1 ]; then
-    # if nodes >1 have to offset +1 so the start frame of this task
-    # doesn't overlap with the end frame of the previous task
-    TASK_START_FRAME=$(( OFFSET - TASK_SIZE + START_FRAME + 1 ))
-  elif [ "$NODES" != "1" ]; then
-    TASK_START_FRAME=$(( OFFSET - TASK_SIZE + START_FRAME ))
-  else
-    TASK_START_FRAME=$START_FRAME
-  fi
+UTIL_FILE="$JOB_TMPDIR/utils.rb"
 
-  echo $TASK_START_FRAME
-}
+cat > "$UTIL_FILE" <<UTILS
+#!/usr/bin/env ruby
 
-function end_frame(){
+def output_start_frame
+  puts task_start_frame
+end
 
-  if [ $REMAINDER -ne 0 ] && [ "$PBS_ARRAYID" -eq "$NODES" ]; then
-    # last task picks up the remainder of the frames
-    TASK_END_FRAME=$(( OFFSET + START_FRAME + REMAINDER ))
-  elif [ "$NODES" != "1" ]; then
-    TASK_END_FRAME=$(( OFFSET + START_FRAME ))
+def task_start_frame
+  if nodes == 1
+    start_frame
+  elsif task_size == 1
+    # you have as many nodes as there are tasks
+    array_id
+  elsif array_id > 1
+    start_frame + offset
   else
-    TASK_END_FRAME=$END_FRAME
-  fi
-
-  echo $TASK_END_FRAME
-}
+    start_frame
+  end
+end
+
+def output_end_frame
+  puts task_end_frame
+end
+
+def task_end_frame
+  if nodes == 1
+    end_frame
+  elsif task_size == 1
+    # you have as many nodes as there are tasks
+    array_id
+  elsif array_id > 1 && array_id == nodes
+    # last task gets the remaining frames
+    offset + task_size + remaining_frames
+  elsif array_id > 1
+    offset + task_size
+  else
+    # array id == 1
+    task_size
+  end
+end
+
+def array_id
+  (ENV['PBS_ARRAYID'] || 1).to_i
+end
+
+def offset
+  (array_id - 1) * task_size
+end
+
+def task_size
+  task_size = (total_frames / nodes).to_i
+  task_size.zero? ? 1 : task_size
+end
+
+def total_frames
+  # need +1 here bc we're always starting at the first frame
+  # 20 - 1 = is 20 frames, not 19. even 3-3 is 1 frame (the third)
+  end_frame - start_frame + 1
+end
+
+def nodes
+  (ENV['NODES'] || 1).to_i
+end
+
+def start_frame
+  frame = (ENV['START_FRAME'] || 1).to_i
+  frame <= 0 ? 1 : frame
+end
```

Comment:
I need to confirm that 1 is the first frame and not 0.

Reply:
0 does seem to be valid frame number, so that's wrong to force it to be greater than 1.

Reply:
Fixed.